### PR TITLE
feat(stream-logs): stream logs to file for job and flow commands

### PR DESCRIPTION
**Pull request by krishna-acondy**
## Issue

Fixes https://github.com/sasjs/cli/issues/587.

## Intent

Stream logs to a file as a job or flow is executing.

## Implementation

* Bumped adapter version.
* Updated `PollOptions` for flow and job commands.
* Only fetched the log after job execution if it hasn't been streamed already.

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
image
- [x] Any new functionality has been unit tested.
- [x] All unit tests are passing (`npm test`).
- [x] All CI checks are green.
- [x] JSDoc comments have been added or updated.
- [x] Reviewer is assigned.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,10 +50,10 @@
     }
   },
   "dependencies": {
-    "@sasjs/adapter": "2.8.14",
+    "@sasjs/adapter": "2.9.0",
     "@sasjs/core": "2.35.4",
     "@sasjs/lint": "1.11.2",
-    "@sasjs/utils": "2.24.0",
+    "@sasjs/utils": "^2.25.4",
```
Comment (saadjutt01) [inline package.json:56]: `^` is present.

**Review by saadjutt01 — CHANGES_REQUESTED** [review]

**Review by allanbowe — APPROVED** [review]

**Comment by allanbowe** [comment]
@all-contributors add @krishna-acondy  for code, infra

**Review by YuryShkoda — APPROVED** [review]

**Review by saadjutt01 — APPROVED** [review]

**Comment by krishna-acondy** [comment]
I'd like to get https://github.com/sasjs/adapter/pull/486 in and update the adapter here before merging this PR.

File: src/commands/job/execute.ts [hunk 2]
```diff
@@ -68,7 +89,7 @@ export async function execute(
     )
   }
 
-  const contextName = getContextName(target, returnStatusOnly)
+  const contextName = await getContextName(target, returnStatusOnly)
```
Comment (krishna-acondy) [inline src/commands/job/execute.ts:92]: @allanbowe / @YuryShkoda I don't know how this was _ever_ working before.

Comment (allanbowe) [reply src/commands/job/execute.ts:92]: clearly we need better tests!

**Review by allanbowe — APPROVED** [review]

**Pull request merged**